src/pages/index.js

```diff
@@ -2,102 +2,193 @@ import "./index.css";
+      });
+      profileModal.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => profileModal.renderLoadingModal(false));
+});
 
-/* -------------------------------------------------------------------------- */
-/*                               Event Listeners                              */
-/* -------------------------------------------------------------------------- */
-const openProfileEditButton = document.querySelector(".profile__edit-button");
 openProfileEditButton.addEventListener("click", () => {
-  const input = userInfo.getUserInfo();
-  profileModal.setInputValues(input);
-  editFormValidator.resetValidation();
+  const currentUserData = userInfo.getUserInfo();
+  profileModal.setInputValues({
+    title: currentUserData.name,
+    description: currentUserData.about,
+  });
   profileModal.open();
 });
 
+profileModal.setEventListeners();
+
 // add new card
 addNewCardButton.addEventListener("click", () => {
   addCardModal.open();
 });
 
 const profileName = ".profile__title";
 const profileDescription = ".profile__description";
+const profileAvatar = ".profile__image";
+
 const userInfo = new UserInfo({
-  userName: profileName,
-  userJob: profileDescription,
+  nameSelector: profileName,
+  aboutSelector: profileDescription,
+  avatarSelector: profileAvatar,
+});
+
+const deleteModal = new PopupConfirm("#remove-card-modal");
+deleteModal.setEventListeners();
+
+function handleDeleteModal(card) {
+  deleteModal.setDeleteFunction(() => {
+    deleteModal.renderLoadingModal(true);
+    api
+      .handleDeleteCard(card._id)
+      .then(() => {
+        card.removeCard();
+        deleteModal.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => deleteModal.renderLoadingModal(false));
+  });
+  deleteModal.open();
+}
+const formValidators = {};
+
+const enableValidation = (config) => {
+  const formList = Array.from(document.querySelectorAll(config.formSelector));
+  formList.forEach((formElement) => {
+    const validator = new FormValidator(config, formElement);
+    const formName = formElement.getAttribute("name");
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+
+function openPreviewModal(cardData) {
+  imageModal.open(cardData);
+}
+
+enableValidation(config);
+formValidators["avatar-form"].disableSubmitButton();
+
+function handleCardLike(card, cardId, isLiked) {
+  api
+    .changeLikeStatus(cardId, isLiked)
+    .then((updatedCard) => {
+      card.setIsLiked(updatedCard.isLiked);
+    })
+    .catch((err) => console.error(err));
+}
+
+function renderCardsAfterUserInfo() {
+  return Promise.all([api.getInitialCards(), api.getUserinfo()]).then(
+    ([cards, userData]) => {
+      cardSection.renderItems(cards);
+      userInfo.setUserInfo({
+        name: userData.name,
+        about: userData.about,
+      });
+      userInfo.setAvatar({
+        avatar: userData.avatar,
+      });
+    }
+  );
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**